test/sequential/test-net-connect-emfile.js

```
@@ -0,0 +1,69 @@
+'use strict';
+
+const net = require('net');
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+
+console.log('Test to check the error message format during EMFILE error');
+
+if (process.platform === 'win32') {
+  console.log('Skipping test, no RLIMIT_NOFILE on Windows.');
+  return;
+}
+
+var totalErrorCount = 0;
+
+const reactToError = function() {
+  totalErrorCount += 1;
+  if (totalErrorCount === 2) {
+    console.log('Test OK');
+    process.exit(0);
+  }
+};
+
+// Server listens
+net.createServer().listen(common.PORT);
+
+// Exhaust the file descriptors
+try {
+  while (true) {
+    fs.openSync(__filename, 'r');
+  }
+} catch (err) {
+  assert(err.code === 'EMFILE' || err.code === 'ENFILE');
+}
+
+// Now connect to the server. We should get EMFILE error from `getaddrinfo`
+const client1 = net.connect({
+  port: common.PORT
+});
+
+client1.on('error', function(er) {
+  assert.equal(er.message, 'getaddrinfo EMFILE');
+  assert.equal(er.code, 'EMFILE');
+  assert.equal(er.syscall, 'getaddrinfo');
+  assert.equal(er.port, common.PORT);
+  reactToError();
+});
+
+// Again connect to the server with the host name. We should get EMFILE error
+// from `connect`
+const client2 = net.connect({
+  host: '127.0.0.1',
+  port: common.PORT
+});
+
+client2.on('error', function(er) {
+  assert.equal(er.message, 'connect EMFILE 127.0.0.1:' + common.PORT);
+  assert.equal(er.code, 'EMFILE');
+  assert.equal(er.syscall, 'connect');
+  assert.equal(er.port, common.PORT);
+  reactToError();
+});
+
+process.on('uncaughtException', function(er) {
+  // Unexpected error. We should fail here.
+  assert.ok(false);
+  process.exit(0);
+});
```

This is unneeded, just let the exception bubble up.